config/config-network.yaml

```diff
@@ -102,6 +102,11 @@ data:
     # We strongly recommend keeping namespace part of the template to avoid domain name clashes
     # Example '{{.Name}}-{{.Namespace}}.{{ index .Annotations "sub"}}.{{.Domain}}'
     # and you have an annotation {"sub":"foo"}, then the generated template would be {Name}-{Namespace}.foo.{Domain}
+    # {{.Unique}} variable can be used as an another option to avoid domain name clashes.
```

Comment:
The guarantees that {{.Unique}} provides does not align with the name "unique" or this comment.  

I do not have a problem with providing the option for an operator to inject randomness into the domainTemplate, but we should be clear that this still may produce conflicts/clashes much like removing {{.Namespace}} in the template opens up the possibility of clashes.

I tested what happens when domains clash due to changes in the domainTemplate. All clashing Services report as healthy and all of the dependent istio objects get created, but only one of them receives all of the traffic. A clash will likely not be obvious or easily diagnosed by operators or users.

Can you update the name of the field to something like: {{.Random}} or {{.RouteHash}}? Also can you update the comment to remove the sections mentioning that "users don't have to worry about collisions".

Reply:
I like `Random`

Reply:
While it is still not clear to me that this solves the original issue, I was thinking a bit too narrowly that since collisions can already be introduced by operators (through removing namespace) that adding this doesn't make things worse. 

Thinking about this further I am also a bit nervous about the upgrade and maintainability story here. Once checked in we will be unable to change the algorithm used to generate the 12 characters as the code is relying on stable output across both Route updates and Knative upgrades while not persisting it anywhere. I don't want to be stuck with 12 characters of a sha256 of the UID as our algorithm here.

Reply:
I think all we need to use as the input value is something that will remain static for the life of the KnService and be unique across all KnServices.  Would using the config's UID be better? I was going to say KnService's UID but I think it's possible for there to not be a KnService but config is required.

Reply:
I concur that we need something that will remain static, if KnRoute's UID is not that good, I  can refactor this using KnService's UID or Config's UID? Or another approach?

Reply:
Let's go with the Config's UID

Reply:
The problem is not the source of the UUID. The problems are:

1. We are shortening a UUID and removing entropy making it more likely to conflict.
2. Conflicts are not detected by the system and behave poorly. One of the two that conflict will respond, but which responds is non-deterministic.
3. The randomness added to the end does not take into account or detect the 63 character limit for DNS names which can also lead to unexpected behavior.
4. We have no way to change this algorithm in future Knative versions as we are not storing the computed value anywhere. Changing the algorithm would change your endpoint for all your Knative Services at one time upon upgrade. There are many reasons which we may want to update how we compute this randomness.